### PR TITLE
refactor: remove redundant embeded interface in kube.Client

### PR DESCRIPTION
**Pull request by drivebyer**
**Please provide a description of this PR:**

The K8s client which embeded in `kube.Client` make the interface less scalable.

In detail, this PR remove the embeded `kubernetes.Interface` of interface `Client` in **pkg/kube/client.go**. Use `Kube() kubernetes.Interface` instead.

### CODE REVIEW

**Comment by costinm** [comment]
Any details ? Curious.

**Comment by SpecialYang** [comment]
What bug does this pr fix?

**Comment by drivebyer** [comment]
hi, @SpecialYang @costinm 
This PR remove the embeded `kubernetes.Interface` of interface `Client` which in pkg/kube/client.go.

File: pkg/kube/client.go [hunk 1]
```diff
@@ -99,8 +99,6 @@ const (
 // clients using a shared config. It is expected that all of Istiod can share the same set of clients and
 // informers. Sharing informers is especially important for load on the API server/Istiod itself.
 type Client interface {
-	// TODO: stop embedding this, it will conflict with future additions. Use Kube() instead is preferred
-	kubernetes.Interface
```
Comment (hzxuzhonghu) [inline pkg/kube/client.go:103]: For reviewers, this is the only change, others are all related caller

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM

**Comment by drivebyer** [comment]
PTAL, @hzxuzhonghu @costinm

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Review by douglas-reid — APPROVED** [review]

**Comment by drivebyer** [comment]
new commits comes from REBASE

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**